src/World.cc

```diff
@@ -569,6 +578,69 @@ Frame *World::FrameByName(const std::string &_name)
       static_cast<const World*>(this)->FrameByName(_name));
 }
 
+/////////////////////////////////////////////////
+uint64_t World::JointCount() const
+{
+  return this->dataPtr->joints.size();
+}
+
+/////////////////////////////////////////////////
+const Joint *World::JointByIndex(const uint64_t _index) const
+{
+  if (_index < this->dataPtr->joints.size())
+    return &this->dataPtr->joints[_index];
+  return nullptr;
+}
+
+/////////////////////////////////////////////////
+Joint *World::JointByIndex(uint64_t _index)
+{
+  return const_cast<Joint*>(
+      static_cast<const World*>(this)->JointByIndex(_index));
+}
+
+/////////////////////////////////////////////////
+const Joint *World::JointByName(const std::string &_name) const
+{
+  auto index = _name.rfind("::");
+  if (index != std::string::npos)
+  {
+    const Model *model = this->ModelByName(_name.substr(0, index));
+    if (nullptr != model)
+    {
+      return model->JointByName(_name.substr(index + 2));
+    }
+
+    // The nested model name preceding the last "::" could not be found.
```

Comment:
Now that `::` is reserved, we can just return `nullptr`.

Reply:
https://github.com/gazebosim/sdformat/pull/1117/commits/92c39e7591a651c21bc3dbe76c19662dd6b47e3d